types/lodash/lodash-tests.ts

```diff
@@ -5651,6 +5651,14 @@ fp.now(); // $ExpectType number
     _.mapValues(abcObject); // $ExpectType AbcObject
     _.mapValues(abcObjectOrNull); // $ExpectType Partial<AbcObject>
 
+    // $ExpectType { a: number; b: number; c: number; }
+    _.mapValues({ a: 1, b: 1, c: 1 }, (value, key, collection) => {
```

Comment:
The test failure is really weird - it looks like it's not getting the right types for `value` and `collection`. If I modify this code to explicitly pass the type parameters, it works file (e.g. `_.mapValues<{ a: number, b: number, c: number }, number>({ a: 1, b: 1, c: 1 }, (value, key, collection) => {`).

Also, explicitly setting the callback parameter types seems to work, so maybe you should do something like this:

```ts
// $ExpectType { a: string; b: string; c: string; }
_.mapValues({ a: 1, b: 1, c: 1 }, (value: number, key: string, collection: { a: number, b: number, c: number }) => "");
```

Reply:
Looks like the test failure is caused by a typescript issue i logged a while ago: TypeScript#21525. Looks like the typescript team has no intention of fixing this.